tests/powder/correction_test.py

```diff
@@ -315,3 +319,43 @@ def test_apply_lorentz_correction_needs_coords():
     )
     with pytest.raises(KeyError):
         apply_lorentz_correction(da)
+
+
+def test_pdf_structure_factor_needs_q_coord():
+    da = sc.DataArray(sc.ones(sizes={'Q': 3}))
+    r = sc.array(dims='r', values=[2, 3, 4, 5.0])
+    with pytest.raises(KeyError):
+        compute_pdf_from_structure_factor(
+            da, sc.scalar(1.0, unit='1/angstrom'), r, use_filter=False
+        )
+
+
+@pytest.mark.parametrize('use_filter', [(False,), (True,)])
+def test_pdf_structure_factor(use_filter):
+    da = sc.DataArray(
+        sc.ones(sizes={'Q': 3}, unit='angstrom**4'),
+        coords={'Q': sc.array(dims='Q', values=[0, 1, 2, 3.0], unit='1/angstrom')},
+    )
+    r = sc.array(dims='r', values=[2, 3, 4, 5.0], unit='angstrom')
+    v = compute_pdf_from_structure_factor(
+        da, sc.scalar(1.0, unit='1/angstrom**3'), r, use_filter=use_filter
+    )
+    assert v.data.unit == '1/angstrom'
+    sc.testing.assert_identical(v.coords['r'], r)
+
+
+def test_pdf_structure_factor_result_unchanged():
+    # Note: bogus data
+    da = sc.DataArray(
+        sc.array(dims='Q', values=[1, 2, 4.0], unit='angstrom**4'),
+        coords={'Q': sc.array(dims='Q', values=[0, 1, 2, 3.0], unit='1/angstrom')},
+    )
+    r = sc.array(dims='r', values=[2, 3, 4, 5.0], unit='angstrom')
+    v = compute_pdf_from_structure_factor(
+        da, sc.scalar(1.0, unit='1/angstrom**3'), r, use_filter=True
+    )
+    sc.testing.assert_allclose(
+        v.data,
+        sc.array(dims='r', values=[-4.25706, -0.528801, -5.29488], unit='1/angstrom'),
```

Comment:
Where do those numbers come from?

Reply:
They're what the routine returns for the test-data in the test right now.
It doesn't mean anything, it just checks if the result has changed.